activerecord/test/config.example.yml

```diff
@@ -94,9 +94,11 @@ connections:
     arunit:
       database: <%= FIXTURES_ROOT %>/fixture_database.sqlite3
       timeout:  5000
+      strict:   true
```

Comment:
Do these lines do anything, since this was done via a config in the end?

Reply:
I can remove in https://github.com/rails/rails/pull/45372

Reply:
Hmm okay, I couldn't see anywhere in this PR that referenced this key but maybe I'm missing it.

Reply:
It is used here https://github.com/rails/rails/blob/bdc3e192bb9b7223963d79cf7a15967a88f4bf6f/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L37
But looks like `strict: ConnectionAdapters::SQLite3Adapter.strict_strings_by_default` overrides the value passed via `database.yml` (I identified the mentioned problems linked in the previous comment before introduced this logic). 
I would highly appreciate if you would fix this too in your PR 😃

Reply:
Hmm, okay, I'm not sure I follow now. It sounds like youre saying you should be able to configure via `database.yml` *or* via `config.active_record.strict_strings_by_default`. What should happen if the two have different values?

Reply:
`strict_strings_by_default` (note the `_by_default` suffix). The value from `database.yml` should override that, if present.

Reply:
Okay, so

`strict_strings_by_default` = true, `database.yml` = false -> strict mode should be false.
`strict_strings_by_default` = true, `database.yml` = true -> strict mode should be true.
`strict_strings_by_default` = true, `database.yml` = (no value) -> strict mode should be true.
`strict_strings_by_default` = false, `database.yml` = true -> strict mode should be true.
`strict_strings_by_default` = false, `database.yml` = false -> strict mode should be false.
`strict_strings_by_default` = false, `database.yml` = (no value) -> strict mode should be false.

Correct?

Reply:
Yes. 
The change should be something like this:
```ruby
config[:strict] = ConnectionAdapters::SQLite3Adapter.strict_strings_by_default unless config.key?(:strict)
db = SQLite3::Database.new(
  config[:database].to_s,
  config.merge(results_as_hash: true)
)
```

Reply:
https://github.com/rails/rails/pull/45373